frontend/src/common/queries/filter.ts

```diff
@@ -220,6 +220,51 @@ export function useFetchDatasetRows(): FetchCategoriesRows<DatasetRow> {
   };
 }
 
+interface CorpusSummary {
+  cellCount: number;
+  cellTypesCount: number;
+  datasets: number;
+  isError: boolean;
+  isLoading: boolean;
+}
+
+export function useCorpusSummary(): CorpusSummary {
+  const datasetRows = useFetchDatasetRows();
+  const summary = datasetRows.rows.reduce(
+    (acc, row) => {
+      const newCellTypes = row.cell_type.reduce((acc1, cellType) => {
+        if (acc.cellTypes.indexOf(cellType.ontology_term_id) === -1) {
+          acc1.push(cellType.ontology_term_id);
+        }
+        return acc1;
+      }, new Array<string>());
+
+      const isPrimaryDataset = row.is_primary_data === IS_PRIMARY_DATA.PRIMARY;
+
+      const newCellCount =
+        acc.cellCount + (isPrimaryDataset ? row.cell_count ?? 0 : 0);
+
+      return {
+        ...acc,
+        cellCount: newCellCount,
+        cellTypes: acc.cellTypes.concat(newCellTypes),
```

Comment:
discussion: any special reason to use `.concat()` over `...`?
```suggestion
        cellTypes: [...acc.cellTypes, ...newCellTypes],
```

The article seems to say only use `concat` when the array is large:
https://www.educative.io/answers/spread-operator-vs-arrayprototypeconcat-in-javascript